cloudinit/sources/helpers/azure.py

```diff
@@ -752,7 +752,7 @@ def build_report(
         status: str,
         substatus=None,
         description=None,
-    ) -> str:
```

Comment:
there's a cascading change that should follow this change. post_health_report should now take bytes instead of str, the http_with_retries should also switch data from str to bytes

Reply:
Turns out the decorator prevented mypy from doing it's job.  Added type info for the decorator to fix it.